File: product-service/src/main/java/com/intexsoft/webshop/productservice/exception/notfound/ResourceNotFoundException.java
package com.intexsoft.webshop.productservice.exception.notfound;

import com.intexsoft.webshop.productservice.exception.HttpStatusException;
import com.intexsoft.webshop.productservice.exception.Resource;
import org.springframework.http.HttpStatus;

public abstract class ResourceNotFoundException extends HttpStatusException {
    private static final String MSG_PATTERN = "%s with id = %d not found";

    protected ResourceNotFoundException(Resource resource, Long resourceId) {
        super(String.format(MSG_PATTERN, resource.name(), resourceId), HttpStatus.NOT_FOUND);
    }
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/exception/notfound/ResourceNotFoundException.kt
package com.intexsoft.webshop.productservicekt.exception.notfound

import com.intexsoft.webshop.productservicekt.exception.HttpStatusException
import com.intexsoft.webshop.productservicekt.exception.Resource
import org.springframework.http.HttpStatus

abstract class ResourceNotFoundException(resource: Resource, resourceId: Long) :
    HttpStatusException("${resource.name} with id = $resourceId not found", HttpStatus.NOT_FOUND)